jetty-core/jetty-http/src/main/java/org/eclipse/jetty/http/HttpParser.java

```diff
@@ -422,6 +422,11 @@ public State getState()
         return _state;
     }
 
+    public boolean hasContent()
+    {
+        return _endOfContent != EndOfContent.NO_CONTENT;
+    }
```

Comment:
I needed this new method because the HttpParser goes to CONTENT state, even when it knows there is NO_CONTENT.   This is because we still send an endContent event before the endMessage event, even for no content.